### PR TITLE
Override #hash from MCDirectoryRepository which assumes a FileDirectory

### PR DESCRIPTION
**Pull request by tom95**
Note that in Pharo this is not necessary as MCDirectoryRepository has instead been adapted.

We could have also changed Monticello's MCDirectoryRepository, however I'm not entirely sure what would happen to all existing repositories in people's images if we change the hash.

### CODE REVIEW

File: MonticelloTonel-Core.package/TonelRepository.class/instance/hash.st [hunk 1]
```diff
@@ -0,0 +1,4 @@
+accessing
```
Comment (j4yk) [inline MonticelloTonel-Core.package/TonelRepository.class/instance/hash.st]: Doesn't #hash usually belong into the 'comparing' protocol? In Object it is in 'comparing'.

**Review by j4yk — APPROVED** [review]
It would have been nice to include in the commit message why this change is necessary or warranted.

The hash method should be recategorized, but this can be done after integrating.

Overall the change looks correct to me, since #= is based on #species and #description, of which the latter is overridden in TonelRepository to rely on an FSReference in the directory variable (and nothing else). The new hash method is based on the same.

**Comment by tom95** [comment]
Oups, indeed, I missed the categorization, thank you for the notice! I added a commit to fix this.

I'd be great if whoever will merge this could add @j4yk's rationale for this change (next to it being the obvious fix to make tonel in Squeak compatible with the assumptions from Pharo's codebase as outlined in my commit message)

**Comment by j4yk** [comment]
Integrated in da70814.

**Pull request closed without merging**